protocol/x/vault/keeper/orders.go

```diff
@@ -95,24 +98,38 @@ func (k Keeper) RefreshVaultClobOrders(ctx sdk.Context, vaultId types.VaultId) (
 			)
 		}
 	}
-
 	// Place new CLOB orders.
 	ordersToPlace, err := k.GetVaultClobOrders(ctx, vaultId)
 	if err != nil {
 		log.ErrorLogWithError(ctx, "Failed to get vault clob orders to place", err, "vaultId", vaultId)
 		return err
 	}
-	for _, order := range ordersToPlace {
+
+	for i, order := range ordersToPlace {
 		err := k.PlaceVaultClobOrder(ctx, order)
 		if err != nil {
 			log.ErrorLogWithError(ctx, "Failed to place order", err, "order", order, "vaultId", vaultId)
 		}
+
 		vaultId.IncrCounterWithLabels(
 			metrics.VaultPlaceOrder,
 			metrics.GetLabelForBoolValue(metrics.Success, err == nil),
 		)
-	}
 
+		// Send indexer order replace messages.
```

Comment:
are vault orders guaranteed to be order replacements? Seems like your code is assuming such?

Reply:
what are the conditions of a replacement? e.g.
1. do `subticks` and `size` have to be equal?
2. what happens if existing order is partially/fully filled?

Reply:
This is the flow with this PR:
1. Vault sends order replacement message to ender
2. Ender removes old order from and writes new order to DB
3. Ender forwards order replacement message to vulcan
4. Vulcan removes old order from and adds new order to Orders Cache. It also removes old order's unfilled amount from OrderbookLevelsCache. If price of old order and new order is different, it sends message to socks with the updated orderbook price level size. If price is the same, message to socks is not sent here.
5. We expect the order update message following the order replacement to update the orderbook price level from the filled amount of the new vault order.

Reply:
Got it. If `subticks` is the same but `size` changes, does vulcan still send message to socks?

Reply:
Not from the order replace handler. Vulcan will send message to socks from the order update message we expect to follow